### ISSUE TITLE
Initialization location for rxloadie, conductor, retrofit

### ISSUE THREAD

**Issue by ersin-ertan** [evant/loadie#2]
Say for a reset password screen, using conductor, rxjava1, retrofit, if you init the loader on reset button press, which then uses the editText text as the recovery email for retrofit. On rotation will not persist the response because the loader is unInit. However if your init in onCreateView, your retrofit call is init with empty text. 

> The given observable must be cold since it's not expected to run until `@link #start()` is called.

Does this imply that the retrofit observable is hot?
I made a few attempts but could not get the configuration correct.

**Comment by evant** [comment]
Can you post an example of what you are trying to do? Not sure I really understand from your description.

**Comment by ersin-ertan** [comment]
[Example code](https://github.com/ersin-ertan/Loadie/tree/master/app/src/main/java/com/nullcognition/loadie)

NoUserInput is great and works when you want to get some data, and persist it for the lifetime of the session, which includes rotations.

However, when you require that user input is passed to the api call, initializing the loader onClick will init your input data to the loader correctly, but will not be persisted on rotation because initialization requires the onClick action. Else initialize in onCreate like NoUserInput, but you are stuck with the default input for the initialized loader.

**Comment by evant** [comment]
`init()` should be always called when it's created. Otherwise it won't re-attach on rotation. For user input, you are using `Observable.just()` this will take the value *at that time* which is not what you want. You should use `Observable.defer()` or `Observable.fromCallable()` instead.

**Comment by ersin-ertan** [comment]
Ok, great. I've got that to work. However, for WithUserInput even though the initialization is in `onCreateView`, upon the first onClick to `startLoader` my callback is null on the loader, requiring this:

```java
  @OnClick(R.id.button) void startLoader() {

    if (!loader.hasResult() && !loader.isRunning()) {
      if (loader.getCallbacks() == null) loader.setCallbacks(stringMyLoaderCallbacks);
      loader.start();
    }
```

**Comment by evant** [comment]
Hm you are right, there appars to be a bug where `loaderManager.start()` doesn't attach the callbacks correctly.

Also a couple of notes:
1) Id's are scopped to the Controller so you don't need to make sure they are globally unique.
2) You don't need to manually call `loaderManager.detach()` when using the Controller provider, the loaderManager lifecycle is handled for you.

**Comment by ersin-ertan** [comment]
Ok, I'll keep that in mind. 

The reason I was calling `loaderManager.detach()` was because after onClick, then after a rotation, for `WithUserInput` the `loader` object was not null, and would cause the following crash:

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.nullcognition.loadie/com.nullcognition.loadie.MainActivity}: java.lang.IllegalStateException: Loader me.tatarka.loadie.RxLoader@281b5db already has callbacks. Make sure you are using unique ids and the LoaderManager was properly detached.
                                                                          
Caused by: java.lang.IllegalStateException: Loader me.tatarka.loadie.RxLoader@281b5db already has callbacks. Make sure you are using unique ids and the LoaderManager was properly detached.
```
@evant 
So the order of the controller creation matters too. I've tried them in separation, then together, and then switched the order of creation in MainActivity, and it seems that the first controller created will get it's constructor called again, which is why there is conflicting functionality. This might get more complicated with controller backstacks, nested controllers(which is where my problem was first discovered), and pager controllers.

**Comment by evant** [comment]
Ah, you should be initializing the loaders in the constructor instead of onCreateView. Conductor has a bit of a different lifecycle so the callback can stay alive even if the views are destroyed and recreated. I wonder if there is a way to have this make more sense.

**Comment by evant** [comment]
Note: just pushed 0.2 which should fix the missing callback issue.

**Comment by ersin-ertan** [comment]
Is there usually a delay between the push, because Gradle's failing to resolve.

http://search.maven.org/#search%7Cga%7C1%7Cme.tatarka.loadie

I'll close the issue after functionality is tested, good team work on locating the issue @evant and thanks for the timely response.

**Comment by evant** [comment]
Yeah there is, maven cental can take a couple hours sometimes.

**Comment by evant** [comment]
If you don't want to wait, you can clone the project, do `./gradlew install` and add `mavenLocal()` to your repositories block.

**Comment by ersin-ertan** [comment]
Using 0.2 the callback problem is fixed.

If a controller is created, the loader is started, result is success, device rotation, if the controller is recreated the constructor is called with: 
```
loaderManager = LoaderManagerProvider.forController(this);
loader = loaderManager.init(
```
the data is not preserved, because conductor has recreated the controller. However, if you onClick and `.start()`, result is success, device rotation any amount of times after, the constructor is not called because the controller is not recreated and the loader works as expected.

After the first rotation, for whatever reason, conductor is recreating the controller, which is not binding to the loader that had its result from the first start press prior to rotation.

- See updated commit for sample

**Comment by evant** [comment]
Hm, I'll take a look. It's certainly true that the current implementation assumes the loader should live as long as the controller.

**Comment by ersin-ertan** [comment]
This will be tricky, how do you differentiate between a conductor initiated controller recreation for the lifetime of _session_(data persisted) vs. a programmatic/manual pop-push, in which a new _session_(data not persisted)is started
